src/workerd/jsg/promise.h

```diff
@@ -496,6 +496,14 @@ class Promise {
 
     using Type = RemovePromise<Result>;
 
+    // TODO(soon): It appears that in some cases the call to Then here can fail
+    // somewhat silently... returning an empty MaybeLocal handle without actually scheduling
+    // an exception on the isolate. When that happens, isolate->IsExecutionTerminating()
+    // also ends up being true. The result is that the check wrapping the Then below
+    // will throw a JsExceptionThrown without an actual exception being scheduled.
+    // We need to investigate further why this happens. In the meantime, code that
+    // hits such an error should wrap itself in a js.tryCatch which will handle the
+    // empty exception case, converting it into an "Unknown error".
```

Comment:
If execution is terminating, it is important that `tryCatch()` does NOT catch this, but instead lets the termination unwind the whole stack.